internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -367,6 +374,46 @@ func (r *NumaflowControllerRolloutReconciler) getResourceOperations() (kubeUtil.
 	return ops, cleanup, nil
 }
 
+func (r *NumaflowControllerRolloutReconciler) prepStatusForUpdate(ctx context.Context, controllerRollout *apiv1.NumaflowControllerRollout) error {
+	numaLogger := logger.FromContext(ctx)
+
+	// Try to get the Numaflow Controller Deployment
+	deployment := kubernetes.GenericObject{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "numaflow-controller",
+			Namespace: controllerRollout.Namespace,
+		},
+	}
+	existingDeployment, err := kubernetes.GetCR(ctx, r.restConfig, &deployment, "deployments")
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// This could just mean that the deployment has not yet been created.
+			// Log and return to wait for next reconcile call once the deployment is created triggering the reconciliation.
+			numaLogger.Warnf("Numaflow Controller Deployment not found. It may still have to be created.")
+		}
+
+		return fmt.Errorf("error getting the Numaflow Controller Deployment for NumaflowControllerRollout %s/%s: %v", controllerRollout.Namespace, controllerRollout.Name, err)
+	}
+
+	// Parse the status of the existing Numaflow Controller Deployment
+	existingDeploymentStatus, err := kubernetes.ParseStatus(existingDeployment)
+	if err != nil {
+		return fmt.Errorf("unable to parse status for existing Numaflow Controller Deployment %s/%s: %v", existingDeployment.Namespace, existingDeployment.Name, err)
+	}
+
+	// Set NumaflowControllerRollout CR status Phase and ObservedGeneration only if the Numaflow Controller Deployment has completely reconciled (Generation == ObservedGeneration)
+	if existingDeployment.Generation == existingDeploymentStatus.ObservedGeneration {
+		controllerRollout.Status.MarkDeployed()
+		controllerRollout.Status.SetObservedGeneration(controllerRollout.Generation)
```

Comment:
Is this line needed here? Did you mean to remove it like you did from the other reconcilers?

Reply:
Forgot to remove that. Thanks